airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/state/StateManager.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.relationaldb.state;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.source.relationaldb.CdcStateManager;
+import io.airbyte.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Defines a manager that manages connector state. Connector state is used to keep track of the data
+ * synced by the connector.
+ *
+ * @param <T> The type of the state maintained by the manager.
+ * @param <S> The type of the stream(s) stored within the state maintained by the manager.
+ */
+public interface StateManager<T, S> {
+
+  /**
+   * Retrieves the {@link CdcStateManager} associated with the state manager.
+   *
+   * @return The {@link CdcStateManager}
+   */
+  CdcStateManager getCdcStateManager();
+
+  /**
+   * Retrieves the map of stream name/namespace tuple to the current cursor information for that
+   * stream.
+   *
+   * @return The map of stream name/namespace tuple to the current cursor information for that stream
+   *         as maintained by this state manager.
+   */
+  Map<AirbyteStreamNameNamespacePair, CursorInfo> getPairToCursorInfoMap();
+
+  /**
+   * Generates an {@link AirbyteStateMessage} that represents the current state contained in the state
+   * manager.
+   *
+   * @param pair The {@link AirbyteStreamNameNamespacePair} that represents a stream managed by the
+   *        state manager.
+   * @return The {@link AirbyteStateMessage} that represents the current state contained in the state
+   *         manager.
+   */
+  AirbyteStateMessage toState(final AirbyteStreamNameNamespacePair pair);
```

Comment:
Should we make `final AirbyteStreamNameNamespacePair pair` Optional so that we dont have to pass null for GlobalState and can instead pass Optional.empty?

Reply:
@subodh1810 Good call.  Will change.

Reply:
This has been changed to `Optional`.